internal/log/log.go

```diff
@@ -17,110 +17,96 @@ type Logger interface {
 }
 
 func init() {
-	defaultLogger = &stdLogger
+	logger = standard
+	SetLogLevel(INFO)
 }
 
+type Level int
+
 // Allows the ordering of severity to be checked
 const (
-	FATAL = 60
-	ERROR = 50
-	WARN  = 40
-	INFO  = 30
-	DEBUG = 20
-	TRACE = 10
+	TRACE = Level(iota - 2)
+	DEBUG
+	INFO
+	WARN
+	ERROR
+	FATAL
 )
 
 var (
-	defaultLogger Logger
-	logLevel      int = INFO
-	logLock       sync.Mutex
-	IsError       bool = true
-	IsWarn        bool = true
-	IsInfo        bool = true
-	IsDebug       bool = false
-	IsTrace       bool = false
+	logger Interface
+	level  Level
+	lock   sync.Mutex
+	// FATAL level is never disabled
+	IsError bool
+	IsWarn  bool
+	IsInfo  bool
+	IsDebug bool
+	IsTrace bool
 )
 
-func SetLogLevel(level int) {
-	logLock.Lock()
+func SetLogLevel(lvl Level) {
+	lock.Lock()
+	IsTrace = false
+	IsDebug = false
+	IsInfo = false
+	IsWarn = false
+	IsError = false
+	IsTrace = false
+	if lvl != TRACE && lvl != DEBUG && lvl != INFO && lvl != WARN && lvl != ERROR && lvl != FATAL {
+		lvl = INFO
+	}
+	level = lvl
 	switch {
-	case level <= TRACE:
-		logLevel = TRACE
-		IsError = true
-		IsWarn = true
-		IsInfo = true
-		IsDebug = true
+	case level == TRACE:
 		IsTrace = true
+		fallthrough
 	case level <= DEBUG:
-		logLevel = DEBUG
-		IsError = true
-		IsWarn = true
-		IsInfo = true
 		IsDebug = true
-		IsTrace = false
+		fallthrough
 	case level <= INFO:
-		logLevel = INFO
-		IsError = true
-		IsWarn = true
 		IsInfo = true
-		IsDebug = false
-		IsTrace = false
+		fallthrough
 	case level <= WARN:
-		logLevel = WARN
-		IsError = true
 		IsWarn = true
-		IsInfo = false
-		IsDebug = false
-		IsTrace = false
+		fallthrough
 	case level <= ERROR:
-		logLevel = ERROR
 		IsError = true
-		IsWarn = false
-		IsInfo = false
-		IsDebug = false
-		IsTrace = false
-	default:
-		logLevel = FATAL
-		IsError = false
-		IsWarn = false
-		IsInfo = false
-		IsDebug = false
-		IsTrace = false
 	}
-	logLock.Unlock()
+	lock.Unlock()
 }
 
 // Output format should be: "timestamp•LOG_LEVEL•filename.go•linenumber•output"
 func Fatal(format string, args ...interface{}) {
-	defaultLogger.Fatal(format, args...)
+	logger.Fatal(format, args...)
 }
 
 func Error(format string, args ...interface{}) {
 	if IsError {
-		defaultLogger.Error(format, args...)
+		logger.Error(format, args...)
 	}
```

Comment:
Sorry, did not notice this before. Redo these if statements as such:
```
func Error(format string, args ...interface{}){
    if !IsError {
          return
    }
    logger.Error(format,args...)
}
```
This is more idiomatic.  It, also, reduces the indention level for the important statements.

Reply:
Seriously?

Reply:
Seems like making the changes will be faster than trying to discuss.  Will do.